### PR TITLE
Don't merge! - SOC-955 | SOC-956 | SOC-1508 | SOC-1509 | Discussion delete/undelete from popover

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-955
https://wikia-inc.atlassian.net/browse/SOC-956
https://wikia-inc.atlassian.net/browse/SOC-1508
https://wikia-inc.atlassian.net/browse/SOC-1509

### CODE REVIEW

File: front/main/app/mixins/discussion-more-options.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+export default Ember.Mixin.create({
+	actions: {
+		/**
+		 * Delete item and close pop-over
+		 *
+		 * @returns {void}
+		 */
+		delete(item) {
+			this.$('.discussion-more-options').mousedown();
```
Comment (bkoval) [inline front/main/app/mixins/discussion-more-options.js]: I would consider explaining why we need it in a comment :)

**Comment by bkoval** [comment]
LGTM :pager:

File: front/main/app/components/discussion-editor.js [hunk 2]
```diff
@@ -265,6 +276,16 @@ export default Ember.Component.extend(ViewportMixin, {
...
 			}
+		},
+
+		/**
+		 * Triggers on textarea's focus
+		 * @param {Event} event
+		 * @returns {void}
+		 */
+		onFocus(event) {
+			event.preventDefault();
```
Comment (rafalkalinski) [inline front/main/app/components/discussion-editor.js]: So is there still a need of calling 'textareaBlur' method in afterCloseActions?

**Comment by rafalkalinski** [comment]
It is very hard to tell where is the new code here...

**Comment by kvas-damian** [comment]
After updating branches it should be easier ;)

**Comment by rafalkalinski** [comment]
LGTM

File: front/main/app/symbols/main/eye.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg width="16" height="12" viewBox="0 0 16 12" xmlns="http://www.w3.org/2000/svg"><path d="M8 8.73c-1.6 0-2.9-1.3-2.9-2.9C5.1 4.22 6.4 2.9 8 2.9c1.6 0 2.9 1.32 2.9 2.92 0 1.6-1.3 2.9-2.9 2.9M8 0C3.25 0 .22 5.24.1 5.46c-.13.22-.13.5 0 .72.12.22 3.17 5.46 7.9 5.46 4.7 0 7.78-5.23 7.9-5.46.14-.22.14-.5 0-.72C15.8 5.24 12.75 0 8 0" fill="#999" fill-rule="evenodd"/></svg>
```
Comment (bkoval) [inline front/main/app/symbols/main/eye.svg]: Please, add a new line at the EOF.

File: front/main/app/styles/module/_discussion.scss [hunk 4]
```diff
@@ -215,21 +238,46 @@ body.discussions {
...
-		  	margin: 0 auto;
+			margin: 0 auto;
 			width: 40px;
 		}
 		&.toggle-share {
 			height: 20px;
 			width: 18px;
 		}
+		&.delete, &.undelete {
+			width: 16px;
```
Comment (bkoval) [inline front/main/app/styles/module/_discussion.scss]: Alphabetical order :)

File: front/main/app/styles/module/_discussion.scss [hunk 5]
```diff
@@ -215,21 +238,46 @@ body.discussions {
...
+			height: 19px;
+		}
+		&.eye {
+			width: 16px;
```
Comment (bkoval) [inline front/main/app/styles/module/_discussion.scss]: Same here, alphabetical order ;)

File: front/main/app/templates/components/post-detail.hbs [hunk 6]
```diff
@@ -7,11 +7,30 @@
...
+
+	{{#pop-over for=(concat 'post-options-' post.id) on='click' flow='flip'}}
+		<ul class="more-options">
+			<li>
+				{{discussion-delete
+					delete=(action 'delete')
+					deleteMessage=(i18n 'main.delete-post' ns='discussion')
+					undelete=(action 'undelete')
+					undeleteMessage=(i18n 'main.undelete-post' ns='discussion')
+					post=post
```
Comment (bkoval) [inline front/main/app/templates/components/post-detail.hbs]: This one is not alphabetically sorted :palm_tree:

File: front/main/app/templates/components/post-reply.hbs [hunk 7]
```diff
@@ -14,11 +14,30 @@
...
+
+		{{#pop-over for=(concat 'reply-options-' post.id) on='click' flow='flip'}}
+			<ul class="more-options">
+				<li>
+					{{discussion-delete
+						delete=(action 'delete')
+						deleteMessage=(i18n 'main.delete-reply' ns='discussion')
+						undelete=(action 'undelete')
+						undeleteMessage=(i18n 'main.undelete-reply' ns='discussion')
+						post=post
```
Comment (bkoval) [inline front/main/app/templates/components/post-reply.hbs]: https://github.com/Wikia/mercury/pull/1862/files#r49986493

**Comment by bkoval** [comment]
LGTM :page_facing_up:

**Pull request closed without merging**